sklearn/gaussian_process/kernels.py

```diff
@@ -498,13 +499,16 @@ def __call__(self, X, Y=None, eval_gradient=False):
                               for kernel in self.kernels])
 
     def __eq__(self, b):
-        if type(self) != type(b) or len(self.kernels) != len(b.kernels):
```

Comment:
is there a reason you changed this condition? We need type identity here, not sub-type.